client/asset/dcr/dcr.go

```diff
@@ -1366,14 +1367,11 @@ func (dcr *ExchangeWallet) Swap(swaps *asset.Swaps) ([]asset.Receipt, asset.Coin
 		}
 		contracts = append(contracts, contractScript)
 		// Make the P2SH address and pubkey script.
-		scriptAddr, err := dcrutil.NewAddressScriptHash(contractScript, dcr.chainParams)
+		scriptAddr, err := stdaddr.NewAddressScriptHashV0(contractScript, dcr.chainParams)
 		if err != nil {
 			return nil, nil, 0, fmt.Errorf("error encoding script address: %w", err)
 		}
-		p2shScript, err := txscript.PayToAddrScript(scriptAddr)
-		if err != nil {
-			return nil, nil, 0, fmt.Errorf("error creating P2SH script: %w", err)
-		}
+		_, p2shScript := scriptAddr.PaymentScript()
```

Comment:
I'm wondering if it's best to go through an address like we're presently doing to get `p2shScript` for `NewTxOut`, or if we should build the script manually.

```go
contractHash160 := stdaddr.Hash160(contractScript)
// HASH160 <20-byte script hash> EQUAL
p2shScript, err := txscript.NewScriptBuilder().
    AddOp(txscript.OP_HASH160).
    AddData(contractHash160).
    AddOp(txscript.OP_EQUAL).
    Script()
```

Basically the reverse of `txscript.ExtractScriptHash`.

I will say that it's probably less likely to break going through an address though.  What do you think @davecgh about building a `MsgTx` given a redemscript to which a p2sh output should apply?

Reply:
I'd stick with the address here.  I expect that we're going to move to blake256 hashes for the script hash at some point (along with a new script version) because 80-bit security is starting to get near the realm of being too low.  Using the address will allow you to just do `stdaddr.NewAddressScriptHashV1(...)` (once it exists of course) and everything else would just work (assuming you also update it to use the returned version for the output too ofc).

Reply:
I'd also suggest changing all instances that create tx outputs such as `txOut` below to use a helper function that also takes the script version:

e.g.

```Go
// newTxOut returns a new transaction output with the given parameters.
func newTxOut(amount int64, pkScriptVer uint16, pkScript []byte) *wire.TxOut {
	return &wire.TxOut{
		Value:    amount,
		Version:  pkScriptVer,
		PkScript: pkScript,
	}
}

p2shScriptVer, p2shScript := scriptAddr.PaymentScript()
txOut := wire.NewTxOut(int64(contract.Value), p2shScriptVer, p2shScriptVer)
```

Reply:
@itswisdomagain In a different PR I've begun updates for meticulous script version checking and passing.  It involves modifying a bunch of types and exported function signatures, so you can defer broader version related changes.  Definitely do use version when the scope is limited, but don't worry about the whole code base in this regard yet.